core/src/main/clojure/xtdb/operator/scan.clj

```diff
@@ -623,11 +623,18 @@
                              :current-iid-ptr (ArrowBufPointer.)}
                       (at-point-point? scan-opts) (assoc :point-point? true))))))
 
-(defn selects->iid-byte-buffer ^ByteBuffer [selects]
+(defn selects->iid-byte-buffer ^ByteBuffer [selects ^RelationReader params-rel]
   (when-let [eid-select (or (get selects 'xt/id) (get selects 'xt$id))]
     (let [eid (nth eid-select 2)]
-      (when (and (= '= (first eid-select)) (s/valid? ::lp/value eid))
-        (trie/->iid eid)))))
+      (when (= '= (first eid-select))
+        (cond
+          (s/valid? ::lp/value eid)
+          (trie/->iid eid)
+
+          (s/valid? ::lp/param eid)
+          (let [eid-rdr (.readerForName params-rel (name eid))]
+            (when (= 1 (.valueCount eid-rdr))
+              (trie/->iid (.getObject eid-rdr 0)))))))))
```

Comment:
Still quite brittle here. I could also make this work for multiple IIDs here, but it seemed like a lot more work.